### PR TITLE
feat: refactor select chain CLI

### PR DESCRIPTION
**Pull request by tiendn**
### Description
This PR refactors the chain selection process in the CLI to split it into two steps:
- Select between **mainnet** and **testnet**.
- Select the specific chain within the chosen network type.

This change helps to manage long lists of chains more efficiently.

Screenshot 2024-07-10 at 15 24 11

### Affected File
```hyperlane-monorepo/typescript/cli/src/utils/chains.ts```

### Drive-by changes
Move select chain into 2 functions

### CODE REVIEW

File: typescript/cli/src/utils/chains.ts [hunk 1]
```diff
@@ -48,19 +50,36 @@ export async function runMultiChainSelectionStep(
...
+  );
   const choices: Parameters<typeof select>['0']['choices'] = [
     { name: '(New custom chain)', value: NEW_CHAIN_MARKER },
-    new Separator('--Mainnet Chains--'),
-    ...chainsToChoices(mainnetChains),
-    new Separator('--Testnet Chains--'),
-    ...chainsToChoices(testnetChains),
+    new Separator(
+      `--${
+        networkType.charAt(0).toUpperCase() + networkType.slice(1)
```
Comment (jmrossy) [inline typescript/cli/src/utils/chains.ts:79]: Use the `toTitleCase` fn from the `@hyperlane-xyz/utils` package to simplify this

**Review by jmrossy — COMMENTED** [review]
One comment but otherwise lgtm :)

File: typescript/cli/src/utils/chains.ts (hunk 1 above)
Comment (tiendn) [reply typescript/cli/src/utils/chains.ts]: tks, i've fixed

**Review by jmrossy — APPROVED** [review]

**Comment by jmrossy** [comment]
@tiendn this PR is failing in CI but I'm not sure why. Your changes shouldn't have any effect. Can you try merging the latest from `main` into your branch? That may solve it

**Pull request merged**